.github/workflows/terraform-ci.yml

```diff
@@ -0,0 +1,218 @@
+name: terraform-ci
+on:
+  push:
+    branches:
+      - 'main'
+      - 'release/**'
+  pull_request:
+    branches:
+      - 'main'
+      - 'release/**'
+env:
+  CONSUL_LICENSE: ${{ secrets.CONSUL_LICENSE }}
+  HCP_CLIENT_ID: ${{ secrets.HCP_CLIENT_ID }}
+  HCP_CLIENT_SECRET: ${{ secrets.HCP_CLIENT_SECRET }}
+permissions: {}
+jobs:
+  action-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Lint workflow
+        uses: docker://docker.mirror.hashicorp.services/rhysd/actionlint:latest
+        with:
+          # we need to ignore the SC2086 rule to pass unescaped $VARS to the terraform commands
+          args: -ignore SC2086
+  get-go-version:
+    runs-on: ubuntu-latest
+    needs:
+      - action-lint
+    defaults:
+      run:
+        working-directory: ./test/acceptance
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "go-version=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+  go-fmt-and-lint-acceptance:
+    runs-on: ubuntu-latest
+    needs:
+      - get-go-version
+    defaults:
+      run:
+        working-directory: ./test/acceptance
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+      - name: go fmt
+        run: |
+          files=$(go fmt ./...)
```

Comment:
minor: golangci-lint does (or can do) a gofmt check for us, so we could potentially remove this step.

Reply:
Great idea! The `gofmt` linter isn't enabled by default but it can be by passing the `--enable gofmt` args. I'll add this and remove the expliclt `go fmt` step.